algorithms/kernel/dtrees/gbt/classification/gbt_classification_train_dense_default_impl.i

```diff
@@ -352,22 +352,63 @@ services::Status ClassificationTrainBatchKernel<algorithmFPType, method, cpu>::c
         DAAL_CHECK_STATUS(s, (indexedFeatures.init<algorithmFPType, cpu>(*x, &featTypes, par.splitMethod == gbt::training::inexact ? &prm : nullptr)));
     }
 
+    algorithmFPType* ptrWeightFeatImp     = nullptr;
+    algorithmFPType* ptrCoverFeatImp      = nullptr;
+    algorithmFPType* ptrCoverTotalFeatImp = nullptr;
+    algorithmFPType* ptrGainFeatImp       = nullptr;
+    algorithmFPType* ptrGainTotalFeatImp  = nullptr;
+
+    if (par.resultsToCompute & gbt::training::computeWeight)
+    {
+        NumericTable* ptrWeight = res.get(weight).get();
+        WriteOnlyRows<algorithmFPType, cpu> rawW(ptrWeight, 0, 1);
```

Comment:
This doesn't work with all NT except Homogen NumericTable.
WriteOnlyRows - RAII class, so memory will be released after existing "if" section

Better to do smth like this:

```cpp
WriteOnlyRows<algorithmFPType, cpu> rawW;

if (par.resultsToCompute & gbt::training::computeWeight) {
   rawW.init(res.get(weight).get(), 0, 1); 
}
```

Reply:
Fixed